### PR TITLE
BSI Based Changes

### PR DESCRIPTION
**Pull request by BraydonKains**
https://rally1.rallydev.com/#/15545269080d/detail/task/367021260940?fdp=true
These are the changes needed for BSI compatibility. Ignore the branch name as it is no longer reflective of the changes. 😄

### CODE REVIEW

File: src/services/cpd-service-factory.js [hunk 1]
```diff
@@ -1,11 +1,14 @@
...
 export class CpdServiceFactory {
 	static getCpdService() {
 		if (window.data && window.data.fraSettings && window.data.fraSettings.valenceHost) {
-			return CpdService;
+			return new CpdService(false);
+		} else if (window.demo) {
+			return DemoCpdService;
 		}
-		return DemoCpdService;
+		return new CpdService(true);
```
Comment (jwong-d2l) [inline src/services/cpd-service-factory.js]: Would inheritance be a better way to share logic in this guy?

Comment (BraydonKains) [reply src/services/cpd-service-factory.js]: I was having a hard time figuring out the right way to use inheritance for this... Maybe we can whiteboard it offline if you have a better idea.

Comment (jwong-d2l) [reply src/services/cpd-service-factory.js]: the only thing we need to change are the options and the host, right?

Comment (BraydonKains) [reply src/services/cpd-service-factory.js]: Yeah, the rest is the same I think.

Comment (jwong-d2l) [reply src/services/cpd-service-factory.js]: https://stackoverflow.com/questions/5441508/how-to-inherit-static-methods-from-base-class-in-javascript

Comment (jwong-d2l) [reply src/services/cpd-service-factory.js]: I dislike changing the way the service works just to support the changes to those two things.

File: src/components/cpd-record-report.js [hunk 2]
```diff
@@ -218,14 +219,9 @@ class CpdRecordReport extends BaseMixin(LitElement) {
 				<div class="d2l-heading-3">
 					${question}
 				</div>
-				<d2l-html-editor
-					key="${this.getQuestionKey(this.records, question.Id)}"
```
Comment (TomMichielsD2L) [inline src/components/cpd-record-report.js]: We no longer need the getQuestionKey function on line 167

File: src/services/cpd-service-factory.js [hunk 3]
```diff
@@ -1,11 +1,14 @@
+import { BsiService } from '../services/bsi-service';
 import { CpdService } from '../services/cpd-service';
 import { DemoCpdService } from '../services/demo-cpd-service';
 
 export class CpdServiceFactory {
 	static getCpdService() {
 		if (window.data && window.data.fraSettings && window.data.fraSettings.valenceHost) {
 			return CpdService;
+		} else if (window.demo) {
```
Comment (TomMichielsD2L) [inline src/services/cpd-service-factory.js]: Did we already set this demo property on the window somewhere?

Comment (BraydonKains) [reply src/services/cpd-service-factory.js]: Nope, this will break the demo for the time being.

File: src/components/cpd-record-report.js (hunk 2 above)
Comment (BraydonKains) [reply src/components/cpd-record-report.js]: Will get this when I fix linting.

File: src/services/cpd-service-factory.js (hunk 3 above)
Comment (BraydonKains) [reply src/services/cpd-service-factory.js]: Update: In the fixes to the tests I opted to also fix the demos.

File: src/services/bsi-service.js [hunk 4]
```diff
@@ -0,0 +1,18 @@
+import { CpdService } from './cpd-service';
```
Comment (jwong-d2l) [inline src/services/bsi-service.js]: so much cleaner! =)

Comment (BraydonKains) [reply src/services/bsi-service.js]: It took Tom and I spitballing on it for me to remember overriding methods was a thing. 😄

**Pull request merged**